extensions/data-workspace/src/dataworkspace.d.ts

```diff
@@ -115,4 +127,21 @@ declare module 'dataworkspace' {
 		 */
 		element: any;
 	}
+
+	export interface IProjectAction {
```

Comment:
also need to add if a toolbar separator should be there after an action. In the mockups, there's a toolbar separator between publish and target platform

Reply:
there should be an IDashboardAction interface with a type property, and then you can have interfaces like: DashboardButtonAction, DashboardDropdownAction...

Reply:
Sure, I will add toolbar separator option.

As for interface (buttons/dropdowns), I will work on that enhancement separately in another PR. This PR is to just provide a basic skeleton for the dashboard.

Reply:
Added a toolbar separator as a property here. And added one between publish and change target platform.